tutorial/README.md

```diff
@@ -0,0 +1,413 @@
+          .save()
+```
+Listing 5: Writing to Redshift tables
+
+spark-redshift automatically creates a Redshift table with the appropriate schema determined from the table/DataFrame being written. The default behavior is to create a new table and to throw an error message if a table with the same name already exists (corresponding to `SaveMode.ErrorIfExists`)
+
+There are two key points to note in Listing 5-
+
+1. Note the `.withColumnRenamed("eventid", "id")`. This feature if necessary if any of the source table columns names are key words in Redshift (ex. table) then spark-redshift will throw and error. This will be [fixed](https://github.com/databricks/spark-redshift/issues/80) in the next release 
+2. Note how we utilize the modes. The first write utilizes the mode `SaveMode.Overwrite` which means that the table will be dropped and recreated it exists. The second query utilizes `SaveMode.Append` which will create the table if it does not exist but will append to it if it does. The default mode is `SaveMode.ErrorIfExists` which creates the table if it does not exists and throws and error if it does. The last mode is `SaveMode.Ignore` which is same as `SaveMode.Overwrite` if the table does not exist but does nothing if the table exists.
+
+While the above examples used Scala we could have also used SQL. And example of writing to a Redshift table in SQL is shows in Listing 6. The only thing to be aware of in the SQL mode is that SaveMode's are not available and only the default mode (`SaveMode.ErrorIfExists`) is in force.
+
+```sql
+> %sql
+CREATE TABLE redshiftevent
+USING com.databricks.spark.redshift
+OPTIONS (
+dbtable 'redshiftevent',
+tempdir 's3n://redshift-spark/temp/',
+  url 'jdbc:redshift://swredshift.czac2vcs84ci.us-east-1.redshift.amazonaws.com:5439/sparkredshift?user=spark&password=mysecretpass'
+)
+AS SELECT * FROM myevent;
+```
+
+By default the save operation uses the key distribution [style](http://docs.aws.amazon.com/redshift/latest/dg/c_choosing_dist_sort.html) of `EVEN` in Redshift. This can be changed by the use of option parameter `diststyle` and `distkey`. See spark-redshift [documentation](https://github.com/databricks/spark-redshift) for details.
+
+### Under the hood - Save Function ###
+
+The implementation of the Save function is provided in the class `com.databricks.spark.redshift.RedshiftWriter`.   
+
+![](images/savetoredshift.gif)
+
+The sequence of steps involved in performing the save function are-
+
+1. The data from the `DataFrame` instance is written out in parallel to the temporary S3 folder. spark-avro is used to write the data to ensure that it encapsulates the schema.
+2. Next a Redshift [COPY](http://docs.aws.amazon.com/redshift/latest/dg/r_COPY.html) command is created and invoked to load the files created in the temporary S3 folder are loaded it into a Redshift table which respects the value of the `SaveMode` parameter passed.
+
+
+## Integration multiple Datasources ##
```

Comment:
Fixed the header as - Integration with other Data Sources

For the rest if I am referring to the API I will call it Data Sources API. If I am simple referring to a source of data I will call it "datasource". If I want to refer to multiple ones I will call it "data sources". Does that sound good.